Cython/Compiler/PyrexTypes.py

```diff
@@ -3776,32 +3776,57 @@ def can_coerce_from_pyobject(self, env):
             return True
         return False
 
+    def _get_conversion_context_cls_cname(self, cname_format_str):
+        # cname_format_str is passed "prefix", "format", "tags"
+        X = "XYZABC"
```

Comment:
How do we actually know that there won't be more than six template parameters? What happens if a user erroneously adds more?

Reply:
I've changed it now to generate extra arguments T1, T2, T3 etc are needed